### PR TITLE
Add Windows DNS and DHCP server artifact definition

### PR DESCRIPTION
**Pull request by Karneades**
This change is [Reviewable](https://reviewable.io/reviews/forensicartifacts/artifacts/226)

### CODE REVIEW

File: data/windows.yaml [hunk 1]
```diff
@@ -2103,3 +2103,23 @@ urls:
 - 'https://www.symantec.com/security_response/writeup.jsp?docid=2012-020609-4221-99&tabid=2'
 - 'http://www.nirsoft.net/utils/winsock_service_providers.html'
 - 'https://msdn.microsoft.com/en-us/library/windows/desktop/ms739923(v=vs.85).aspx'
+---
+name: WindowsDNSNameServer
+doc: |
+    Windows DNS and DHCP Server Registry Keys.
+
+    Typical location for hijacking the DNS or DHCP servers of a client.
```
Comment (joachimmetz) [inline data/windows.yaml]: > Typical location for hijacking the DNS or DHCP servers of a client.

These are the locations the DNS setting are stored in Windows. I agree that malware can change them but for the artifact itself IMHO less relevant.

Maybe change doc to:
Windows Registry Keys that contain DNS and DHCP settings.

File: data/windows.yaml [hunk 2]
```diff
@@ -2103,3 +2103,23 @@ urls:
 - 'https://www.symantec.com/security_response/writeup.jsp?docid=2012-020609-4221-99&tabid=2'
 - 'http://www.nirsoft.net/utils/winsock_service_providers.html'
 - 'https://msdn.microsoft.com/en-us/library/windows/desktop/ms739923(v=vs.85).aspx'
+---
+name: WindowsDNSNameServer
```
Comment (joachimmetz) [inline data/windows.yaml]: The NS in DNS already refers to Name Sever, maybe rename this to: WindowsDNSSettings

File: data/windows.yaml [hunk 3]
```diff
@@ -2103,3 +2103,23 @@ urls:
...
+    key_value_pairs:
+      - {key: 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Tcpip\Parameters', value: 'NameServer'}
+      - {key: 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters\Interfaces\*', value: 'NameServer'}
+      - {key: 'HKEY_LOCAL_MACHINE\SYSTEM\CurrenControlSet\Services\Dnscache\Parameters', value: 'NameServer'}
+      - {key: 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\services\Tcpip\Parameters\Interfaces\*', value: 'DhcpNameServer'}
+      - {key: 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\services\Tcpip\Parameters\Interfaces\*', value: 'DhcpServer'}
+labels: [System, Network]
+supported_os: [Windows]
+urls: ['https://technet.microsoft.com/en-us/library/dd197418(v=ws.10).aspx']
+
```
Comment (joachimmetz) [inline data/windows.yaml]: please remove trailing white line

**Comment by joachimmetz** [comment]
Thx for the contribution, I have some minor comments / suggestions about some of the wording. When those are addressed these changes are good to go in.

**Comment by Karneades** [comment]
Thanks for the comments. Changed it accordingly.

**Comment by joachimmetz** [comment]
LGTM

**Pull request merged**